### PR TITLE
Makes the energy calculation part of temp share a define.

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
I see share used semi often in our code, might as well document what it does since its not immediately obvious at the first glance.

<details>
<summary>Derivation but with formatting:</summary>

</details>

## Why It's Good For The Game
Maintainability

### CODE REVIEW

**Comment by Ghilker** [comment]
neat

File: code/__DEFINES/atmospherics/atmos_helpers.dm [hunk 1]
```diff
@@ -74,3 +74,31 @@ GLOBAL_LIST_INIT(atmos_adjacent_savings, list(0,0))
...
+ * This is the energy transferred:
+ * W = T' * C1 - W1
+ * W = (W1+W2) / (C1+C2) * C1 - W1
+ * W = (W1C1 + W2C1) / (C1+C2) - W1
+ * W = ((W1C1 + W2C1) - (W1 * (C1+C2))) / (C1+C2)
+ * W = ((W1C1 + W2C1) - (W1C1 + W1C2)) / (C1+C2)
+ * W = (W1C1 - W1C1 + W2C1 - W1C2) / (C1+C2)
+ * W = (W2C1 - W1C2) / (C1+C2)
+ * W = (T2*C2*C1 - T1*C1*C2) / (C1+C2)
+ * W = (C1*C2) * (T2-T1) / (C1+C2)
```
Comment (LemonInTheDark) [inline code/__DEFINES/atmospherics/atmos_helpers.dm:93]: define your terms

**Comment by vincentiusvin** [comment]
damn they added latex support like 20 hours ago

$$\Delta W = \frac{W_1+W_2}{C_1+C_2} \cdot C_1 - W_1$$

$$\Delta W = \frac{W_1C_1+W_2C_1}{C_1+C_2} - W_1$$

$$\Delta W = \frac{W_1C_1+W_2C_1- (W_1\cdot(C_1+C_2))}{C_1+C_2} $$

$$\Delta W = \frac{W_1C_1+W_2C_1- (W_1C_1+W_1C_2)}{C_1+C_2} $$

$$\Delta W = \frac{W_1C_1+W_2C_1 - W_1C_1 - W_1C_2}{C_1+C_2} $$

$$\Delta W = \frac{W_2C_1 - W_1C_2}{C_1+C_2} $$

$$\Delta W = \frac{T_2C_2C_1 - T_1C_1C_2}{C_1+C_2} $$

$$\Delta W = \frac{C_1C_2 (T_2 - T_1)} {C_1+C_2} $$

**Comment by LemonInTheDark** [comment]
Surely there are more places this applies to

**Comment by vincentiusvin** [comment]
Oh cryo uses it too

**Comment by vincentiusvin** [comment]
damn 30d7f9dd1a6e3fa15381316cdccebb93190a462b is messy as hell i rewrote it a bit would appreciate a bit more scrutiny here.

**Pull request merged**